internal/xds/translator/route.go

```diff
@@ -135,6 +140,31 @@ func buildXdsRouteAction(routeName string) *route.RouteAction {
 	}
 }
 
+func buildXdsWeightedRouteAction(httpRoute *ir.HTTPRoute) *route.RouteAction {
+	invalidBackendWeight := uint32(100 * float64(httpRoute.InvalidBackends) / float64(len(httpRoute.Destinations)+httpRoute.InvalidBackends))
```

Comment:
shouldnt the weight take into consideration the weight of each destination ?

Reply:
It absolutely does need to, good catch :+1: